src/rgw/rgw_bucket.cc

```diff
@@ -868,7 +868,15 @@ int RGWBucket::check_bad_index_multipart(RGWBucketAdminOpState& op_state,
   map<string, bool> meta_objs;
   map<rgw_obj_key, string> all_objs;
 
-  RGWRados::Bucket target(store, bucket);
+  RGWBucketInfo bucket_info;
+  RGWObjectCtx obj_ctx(store);
+  int r = store->get_bucket_instance_info(obj_ctx, bucket, bucket_info, NULL, NULL);
```

Comment:
@oritwas How good to take `int` as `int32_t`. 
https://google.github.io/styleguide/cppguide.html#Integer_Types